src/views/cdi-upload-provider/hooks/useCDIUpload.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useRef, useState } from 'react';
+import axios from 'axios';
+
+import { CDIConfigModelRef } from '@kubevirt-ui/kubevirt-api/console';
+import { CDIConfig, getUploadProxyURL } from '@kubevirt-utils/hooks/useCDIUpload/utils';
+import { useK8sWatchResource, WatchK8sResource } from '@openshift-console/dynamic-plugin-sdk';
+
+import { CDI_UPLOAD_URL_BUILDER, UPLOAD_STATUS } from '../utils/consts';
+import { CDIUploadContextProps } from '../utils/context';
+import { DataUpload, UploadDataProps } from '../utils/types';
+
+const resource: WatchK8sResource = {
+  kind: CDIConfigModelRef,
+  isList: false,
+  namespaced: false,
+  name: 'config',
+};
+
+const useCDIUpload = (): CDIUploadContextProps => {
```

Comment:
Already talked about that.
`useCDIUpload` is also used in the disk modal to upload iso files.
It's in `src/utils/hooks/useCDIUpload` and in the near future, we can think of merging these two hooks.
Posted just as memo

Reply:
yes, should be checked in follow up pr